cmd/goal/application.go

```diff
@@ -275,6 +327,9 @@ func getAppInputs() (args [][]byte, accounts []string, foreignApps []uint64, for
 	}
 
 	var encodedArgs []appCallArg
+
+	splitAppArgsByComma()
+
```

Comment:
Going along with my previous comment - this looks like magic.  If the function takes and returns the list, you can show what's going on.